mono/mini/method-to-ir.c

```diff
@@ -10605,8 +10610,42 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				break;
 			}
 
-			*sp++ = handle_box (cfg, val, klass, context_used);
+			if (mono_class_is_nullable (klass) || mini_is_gsharedvt_klass (klass) || context_used) {
+				*sp++ = mini_emit_box (cfg, val, klass, context_used);
+			} else {
+				MONO_INST_NEW (cfg, ins, OP_BOX);
+				ins->type = STACK_OBJ;
+				ins->klass = klass;
+				ins->sreg1 = val->dreg;
+				ins->dreg = alloc_dreg (cfg, val->type);
+
+				switch (val->type) {
+				case STACK_I4:
+				case STACK_I8:
```

Comment:
Hmmmmmmm, how does this handle longs on 32bits? Don't we need a OP_LBOX that takes as input the regpair?

Reply:
Possible.